src/shared/messaging/port-provider.js

```diff
@@ -139,12 +139,12 @@ export class PortProvider {
     const handleRequest = event => {
       const { data, origin, source } = /** @type {MessageEvent} */ (event);
 
-      if (!isMessage(data) || data.type !== 'request') {
+      if (!isMessage(data) || data.type !== 'request' || !data.requestId) {
```

Comment:
Why don't you update `isMessage` to check for the presence of the `requestId` property?

Reply:
That would make sense. I think I didn't initially because `requestId` was an optional property only present on `request` messages initially.